compiler/modules/nback.llvm/llvm.bal

```diff
@@ -566,27 +566,40 @@ public class Builder {
     }
 
     // Corresponds to LLVMBuildGEP
-    public function getElementPtr(PointerValue ptr, Value[] indices,"inbounds"? inbounds=(), string? name = ()) returns PointerValue {
-        if indices.length() > 1 {
-            panic error ("More than one index not supported");
-        }
+    public function getElementPtr(PointerValue ptr, Value[] indices, "inbounds"? inbounds = (), string? name = ()) returns PointerValue {
         BasicBlock bb = self.bb();
         string reg = bb.func.genReg();
         string[] insWords = [];
-        insWords.push(reg);
-        insWords.push("=");
-        insWords.push("getelementptr");
-        insWords.push(typeToString(ptr.ty.pointsTo));
-        insWords.push(",");
-        insWords.push(typeToString(ptr.ty));
-        insWords.push(ptr.operand);
+        insWords.push(reg, "=", "getelementptr");
+        if inbounds != () {
+            insWords.push(inbounds);
+        } 
+        insWords.push(typeToString(ptr.ty.pointsTo), ",", typeToString(ptr.ty), ptr.operand);
+        Type resultType = ptr.ty;
         foreach var index in indices {
             insWords.push(",");
             insWords.push(typeToString(index.ty));
             insWords.push(index.operand);
+            if resultType is PointerType {
+                resultType = resultType.pointsTo;
+            } else {
+                if resultType is ArrayType {
+                    resultType = resultType.elementType;
+                } else if resultType is StructType {
+                    int|error i = int:fromString(index.operand);
```

Comment:
This is a good place to use `checkpanic`.

Reply:
Fixed